### PR TITLE
Megarefactor payment flow

### PR DESCRIPTION
**Pull request by cloudify**
Rely on navigation state to keep track of the payment flow, avoids duplicating the navigation logic (react-navigator and custom).

Todo:

- [x] fix loading states, make them more granular
- [ ] handle back navigation within a payment
- [ ] handle pin authorization within a payment
- [ ] add handling of error states

### CODE REVIEW

File: ts/components/wallet/PaymentMethodsList.tsx [hunk 1]
```diff
@@ -88,8 +61,28 @@ const AddMethodStyle = StyleSheet.create({
...
+        name: I18n.t("wallet.methods.card.name"),
+        maxFee: I18n.t("wallet.methods.card.maxFee"),
+        icon: "io-48-card"
+      },
+      {
+        navigateTo: undefined, // TODO: add route when destination is available @https://www.pivotaltracker.com/story/show/157588719
+        name: I18n.t("wallet.methods.bank.name"),
+        maxFee: I18n.t("wallet.methods.bank.maxFee"),
+        icon: "io-48-bank",
+        onPress: unavailableAlert
```
Comment (fpersico) [inline ts/components/wallet/PaymentMethodsList.tsx]: Cant we rename `navigateTo` to onPressHandler and avoid to have a different onPress for not defined payment method?

if a payment method is implemented
onPressHandler: this.props.navigateToThePaymentMethod

if a payment method is not implemented
onPressHandler: unavailableAlert

File: ts/components/wallet/PaymentMethodsList.tsx [hunk 2]
```diff
@@ -103,7 +96,7 @@ class PaymentMethodsList extends React.Component<Props, never> {
               style={[AddMethodStyle.paymentMethodEntry, WalletStyles.listItem]}
               onPress={() =>
                 itemInfo.item.navigateTo
-                  ? navigate(itemInfo.item.navigateTo)
+                  ? itemInfo.item.navigateTo()
                   : itemInfo.item.onPress
```
Comment (fpersico) [inline ts/components/wallet/PaymentMethodsList.tsx]: Read above

File: ts/components/wallet/PaymentMethodsList.tsx [hunk 3]
```diff
@@ -103,7 +96,7 @@ class PaymentMethodsList extends React.Component<Props, never> {
               style={[AddMethodStyle.paymentMethodEntry, WalletStyles.listItem]}
               onPress={() =>
                 itemInfo.item.navigateTo
-                  ? navigate(itemInfo.item.navigateTo)
+                  ? itemInfo.item.navigateTo()
```
Comment (fpersico) [inline ts/components/wallet/PaymentMethodsList.tsx]: Read above

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a Pivotal story at the beginning of the PR title (see below).
  </td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**